types/src/v0/impls/stake_table.rs

```diff
@@ -475,7 +475,12 @@ impl StakeTableFetcher {
         // which are stored directly in persistence as is.
         // However, this step is taken as a precaution.
         // The vector is already sorted above, so this should be fast.
+        let len_before_dedup = events.len();
         events.dedup();
+        let len_after_dedup = events.len();
+        if len_before_dedup != len_after_dedup {
+            tracing::warn!("Duplicate events found and removed. This should not normally happen.")
```

Comment:
Argh it's annoying we don't get the duplicate events returned. It would be so hard to debug this just with the log message.